frontend/app/(tabs)/home.tsx

```diff
@@ -47,12 +49,12 @@ const Index = () => {
     } finally {
       setIsLoading(false);
     }
-  }, [page, isLoading]);
+  }, [page, isLoading, searchQuery]);
 
   const handleSearch = (query: string) => {
     if (query.length >= 2) {
       loadCards(query);
-    } else {
+    } else if (query.length === 0) {
```

Comment:
Why do we need this? I think is better without as the else statement otherwise cover both 0 and 1

Reply:
We shouldn't need this anymore this should be in the fetch cards method now, handle search should just pass the search to the method